crates/aggchain-proof-core/src/full_execution_proof.rs

```diff
@@ -166,7 +177,7 @@ impl FepInputs {
             #[cfg(target_os = "zkvm")]
             {
                 sp1_zkvm::lib::verify::verify_sp1_proof(
-                    &AGGREGATION_VKEY_HASH,
+                    &self.aggregation_vkey_hash,
```

Comment:
Since the aggchain verification key no longer commits to the aggregation and range vkeys, what enforces the keys are the expected ones? Id the L1 contract now supposed to check the keys specified as public values against some sort of white list?

Reply:
yeah exactly, they are moved as public inputs through the `aggchain_params` which is then propagated as public input of the pessimistic proof through the `aggchain_hash` which is itself re-constructed on the L1.

In other words, if the vkeys used here arent the ones registered on L1, then the `aggchain_hash` would be incorrect because of the `aggchain_params` which wont contain the right vkeys hash

Reply:
I think they won't need to be hardcoded anymore and could be dynamically fetched from L1 contracts, so basically your option 3.

For debugging I think simply logging them to know which ones were considered would be good enough

I'm spinning off this as dedicated issue to track that one 👍

Reply:
cross posting the issue: #189